plugins/API/Controller.php

```diff
@@ -137,4 +137,16 @@ public function listSegments()
 		</table>
 		";
     }
+
+    public function glossary()
+    {
+        Piwik::checkUserHasSomeViewAccess();
+
+        $glossary = new Glossary($this->idSite);
```

Comment:
I see this PR is still a work in progress, but I want to mention that classes that encapsulate application logic should be stateless or immutable service classes and should be obtained through DI. Instead of creating an instance of this type manually here, you should get it through StaticContainer.

To do this, you must:

* Remove the idSite as a constructor parameter and move to individual methods.
* Add constructor arguments for all dependencies in `Glossary` class. Dependencies that aren't set via DI (ie, singletons that are not yet in DI) should be set in the constructor, but not as constructor arguments. (These I'll change manually when getting rid of Singleton/StaticContainer).